celery/worker/pidbox.py

```diff
@@ -48,16 +48,22 @@ def on_message(self, body, message):
             self.reset()
 
     def start(self, c):
-        self.node.channel = c.connection.channel()
-        self.consumer = self.node.listen(callback=self.on_message)
-        self.consumer.on_decode_error = c.on_decode_error
+        if 'transport_options' in c.connection.info():
```

Comment:
You should actually be able to use `if c.connection.supports_exchange_type('fanout'):` now!